src/rust/catnip/runtime/network.rs

```diff
@@ -51,7 +55,7 @@ impl NetworkRuntime for DPDKRuntime {
         // Chain body buffer.
 
         // First, allocate a header mbuf and write the header into it.
-        let mut header_mbuf = match self.mm.alloc_header_mbuf() {
+        let mut header_mbuf: DPDKBuffer = match self.mm.alloc_header_mbuf() {
```

Comment:
I would really enjoy to entirely replace the `HeapBuffer` and `DPDKBuffer` abstractions for just `DemiBuffer`. What are we missing to achieve this? Could you add some TODO for this and create us an issue?